nodejs/week1/avg.js

```diff
@@ -0,0 +1,21 @@
+(function () {
+    try {
+        if (process.argv.slice(2).length) {
+            const numericInput = process.argv.slice(2).filter(x => (!isNaN(x))).length;
+            if (numericInput) {
+                const sum = process.argv.slice(2).map(x => (parseFloat(x))).filter(x => !isNaN(x)).reduce((a, b) => a + b)
+                const avg = parseFloat(sum / numericInput);
+                console.log(`average of numeric elements are ${avg}`)
```

Comment:
For suppose, the sum results in 0 or isNaN eventually avg will be NAN.
To avoid this, One of the solutions is: to check if avg is isNaN.

Reply:
Hi 
Thanks for reviewing 
I have one confusion as I am checking all inputs whether its numeric or not so if all inputs are numeric then sum should be numeric ,is there any need to check avg explicitly NaN or not,please suggest.
and in any case if sum has possibility to be 0 then checking that part by numericInput identifier,please advice if I am understanding something wrong or missing something

Thank You

Reply:
Okay, I got it.
When you have an array of validated numerical data, You can use that array to map() and reduce(). You don't need to use raw data again and filter() it.